### PR TITLE
Resize Bilinear interpolation - Tensor support

### PR DESCRIPTION
**Pull request by fiona-gladwin**
Add U8 resize bilinear interpolation support.

### CODE REVIEW

**Review by r-abishek — CHANGES_REQUESTED** [review]
Please look into these changes

File: src/modules/cpu/host_tensor_augmentations.hpp [hunk 1]
```diff
@@ -5265,4 +5265,356 @@ RppStatus color_cast_i8_i8_host_tensor(Rpp8s *srcPtr,
...
+        Rpp32f srcLocationRow, srcLocationColumn, pixel;
+        Rpp32s srcLocationRowFloor, srcLocationColumnFloor;
+        Rpp32u alignedLength = (dstDescPtr->w / 4) * 4;
+        __m128 pWRatio = _mm_set1_ps(wRatio);
+        __m128 pOne = _mm_set1_ps(1.0);
+        __m128 pWidthLimit = _mm_set1_ps((float)widthLimit);
+        __m128 pChannel = _mm_set1_ps((float) srcDescPtr->c);
+        __m128 p0, p2, p4, p5, p6, p7, pColFloor;
+        __m128i pxColFloor;
+        __m128 pRow[16], pPixels[4];
```
Comment (r-abishek) [inline src/modules/cpu/host_tensor_augmentations.hpp:5337]: Not sure if you are using all 16 and 4 allocated?

File: src/include/cpu/rpp_cpu_simd.hpp [hunk 2]
```diff
@@ -885,4 +893,443 @@ static inline void fast_matmul4x4_sse(float *A, float *B, float *C)
...
+    pTemp[2] = _mm_loadu_ps((float *)(srcPtrBottomRow + loc[2] + 3));
+    pTemp[3] = _mm_loadu_ps((float *)(srcPtrBottomRow + loc[3] + 3));
+    pTemp[4] = _mm_unpacklo_ps(pTemp[0], pTemp[2]);
+    pTemp[5] = _mm_unpacklo_ps(pTemp[1], pTemp[3]);
+    pTemp[6] = _mm_unpackhi_ps(pTemp[0], pTemp[2]);
+    pTemp[7] = _mm_unpackhi_ps(pTemp[1], pTemp[3]);
+    p[9] = _mm_unpacklo_ps(pTemp[4], pTemp[5]);
+    p[10] = _mm_unpackhi_ps(pTemp[4], pTemp[5]);
+    p[11] = _mm_unpacklo_ps(pTemp[6], pTemp[7]);
+    return RPP_SUCCESS;
```
Comment (r-abishek) [inline src/include/cpu/rpp_cpu_simd.hpp:1136]: One line space before all return RPP_SUCCESS;

File: src/include/cpu/rpp_cpu_simd.hpp [hunk 3]
```diff
@@ -885,4 +893,443 @@ static inline void fast_matmul4x4_sse(float *A, float *B, float *C)
...
+    _mm_storeu_si128((__m128i *)(dstRPtr), px1);
+    px1 = _mm_packus_epi16(_mm_packus_epi32(_mm_cvtps_epi32(p[1]), pxZero), pxZero);
+    _mm_storeu_si128((__m128i *)(dstGPtr), px1);
+    px1 = _mm_packus_epi16(_mm_packus_epi32(_mm_cvtps_epi32(p[2]), pxZero), pxZero);
+    _mm_storeu_si128((__m128i *)(dstBPtr), px1);
+
+    return RPP_SUCCESS;
+}
+
+inline RppStatus rpp_store4_f32pln_to_u8pln(Rpp8u* dstPtr, __m128 p)
```
Comment (r-abishek) [inline src/include/cpu/rpp_cpu_simd.hpp:1046]: Is this just for pln1 to pln1? Then change to say f32pln1_to_u8pln1

**Review by r-abishek — CHANGES_REQUESTED** [review]
Please take a look at restructuring and this second round of changes

File: src/modules/cpu/host_tensor_augmentations.hpp [hunk 4]
```diff
@@ -5265,4 +5265,1285 @@ RppStatus color_cast_i8_i8_host_tensor(Rpp8s *srcPtr,
...
+                        rpp_simd_store(rpp_store4_f32pln3_to_u8pkd3, dstPtrRow, pPixels);
+                        dstPtrRow += 12;
+                    }
+                }
+                if(dstDescPtr->layout == RpptLayout::NCHW)
+                {
+                    for (; vectorLoopCount < dstImgSize[batchCount].width; vectorLoopCount++)
+                    {
+                        compute_resize_src_loc(vectorLoopCount, wRatio, widthLimit, srcLocationColumnFloor, weightedWidth, true);
+                        weightedWidth1 = 1 - weightedWidth;
```
Comment (r-abishek) [inline src/modules/cpu/host_tensor_augmentations.hpp:5391]: Compute weightedWidth1 inside too

File: src/modules/cpu/host_tensor_augmentations.hpp [hunk 5]
```diff
@@ -5265,4 +5265,1285 @@ RppStatus color_cast_i8_i8_host_tensor(Rpp8s *srcPtr,
...
+                        Rpp32f m1 = weightedHeight1 * weightedWidth1;
```
Comment (r-abishek) [inline src/modules/cpu/host_tensor_augmentations.hpp:5392]: You could have a templated compute_bilinear_interpolation() just like the compute_bilinear_interpolation_sse() for L5392-L5407 and reuse everywhere

File: src/include/cpu/rpp_cpu_simd.hpp [hunk 6]
```diff
@@ -885,4 +894,477 @@ static inline void fast_matmul4x4_sse(float *A, float *B, float *C)
...
+        *(dstPtr + i) = (Rpp16f)(temp[0][i]);
+        *(dstPtr + 3 + i) = (Rpp16f)(temp[1][i]);
+        *(dstPtr + 6 + i) = (Rpp16f)(temp[2][i]);
+        *(dstPtr + 9 + i) = (Rpp16f)(temp[3][i]);
+    }
+
+    return RPP_SUCCESS;
+}
+
+inline void compute_resize_src_loc_sse(__m128 &dstLoc, __m128 &scale, __m128 &limit, Rpp32u *srcLoc, __m128 &weight, bool hasRGBChannels = false)
```
Comment (r-abishek) [inline src/include/cpu/rpp_cpu_simd.hpp:1344]: Lets move all these compute functions to rpp_cpu_common under the compute section. Use the rpp_cpu_simd only for generalized load/store routine pairs.

**Review by r-abishek — CHANGES_REQUESTED** [review]
Another round of changes. @fiona-gladwin please take a look.

File: src/include/cpu/rpp_cpu_simd.hpp (hunk 6 above)
Comment (fiona-gladwin) [reply src/include/cpu/rpp_cpu_simd.hpp]: Done

**Review by r-abishek — APPROVED** [review]
Looks much better now

**Pull request merged**